src/Microsoft.OpenApi/Validations/ValidationRuleSet.cs

```diff
@@ -185,17 +185,15 @@ IEnumerator IEnumerable.GetEnumerator()
         private static ValidationRuleSet BuildDefaultRuleSet()
         {
             var ruleSet = new ValidationRuleSet();
-            var validationRuleType = typeof(ValidationRule);
+            
+            var ruleTypeProperties = GetValidationRuleTypes();
 
-            var rules = typeof(ValidationRuleSet).Assembly.GetTypes()
-                .Where(t => t.IsClass
-                            && t != typeof(object)
-                            && t.GetCustomAttributes(typeof(OpenApiRuleAttribute), false).Any())
-                .SelectMany(t2 => t2.GetProperties(BindingFlags.Static | BindingFlags.Public)
-                                .Where(p => validationRuleType.IsAssignableFrom(p.PropertyType)));
-
-            foreach (var property in rules)
+            foreach (var property in ruleTypeProperties)
             {
+                if (!typeof(ValidationRule).IsAssignableFrom(property.PropertyType))
```

Comment:
I'm not sure how much perf is a concern here (probably isn't), but I recently learned that `typeof(X)` doesn't get cached locally, so each time through this loop the Type gets re-lookedup. If perf is a concern, keeping a local variable for the Type helps.

Reply:
The `BuildDefaultRuleSet` is called once and cached so the overall cost of this is pretty small but it doesn't hurt to revert this change for a little perf. 😄